SOC-715 As a user on desktop, I want to sign in from a modal

rafalkalinski opened the pull request:
https://wikia-inc.atlassian.net/browse/SOC-715

fraszczakszymon committed:
ADEN-2289 Hop in interstitial

Commit:
ADEN-2312 INCONTENT_PLAYER to 1x1 size

Commit:
CONCF-743 | remove jQuery data('crop') in addition to the DOM attribute

Commit:
CE-2367 replace method isTitleWithNonportableInfobox

Commit:
CONCF-967 | validate video provider for featured items

Commit:
Merge remote-tracking branch 'origin/dev' into CE-2367

fraszczakszymon committed:
ADEN-2289 Switched hidden to isVisible modal logic

Commit:
Merge pull request #8080 from Wikia/CONCF-743

CONCF-743 | remove data-crop when removing image in desktop CC tool

bognix committed:
CONCF-978 restrict permissions to getData and setData to admins only

Grunny committed:
(MAIN-5152) Properly sanitize query when getting WF variables

Grunny committed:
(MAIN-5152) Properly sanitize query when getting WF variables

fraszczakszymon committed:
ADEN-2289 Changed adType to highimpact

Commit:
Merge branch 'dev' of github.com:Wikia/app into dev

Grunny committed:
Merge pull request #8083 from WikiaVolDev/ga-undefined-fix

Fix window.ga being undefined when wgNoExternals is set

Commit:
Merge pull request #8065 from Wikia/INT-124

INT-124: Adding message string for mobile wikia-in-your-lang

Commit:
Use the wiki name being passed in rather than rely on {{SITENAME}}

Commit:
return 404 if the user page does not exist

Commit:
Merge branch 'dev' of github.com:Wikia/app into dev

Commit:
SOC-677 set attributes in UAS after call to User::setGlobalAttribute

bekcunning committed:
Merge pull request #8027 from Wikia/SOC-1096

SOC-1096 Send digest emails from the correct wikia

owend committed:
Merge pull request #8095 from Wikia/MAIN-4970

return 404 if the user page does not exist

Commit:
removing old/dead code

mari-mcmurtrie committed:
INT-158: Upload photo for Japan from Mercury - if anonymous edit is disabled in the wikia, then we also do not allow photo upload

mari-mcmurtrie committed:
INT-158: Add anonymous edit enabled information for Mercury front-end.

Commit:
Pave the way for isBlocked

- Add isBlocked to the controller
- Create \Wikia\Service\User\Auth
- Add minimal tests for isUsernameBlocked functionality

mari-mcmurtrie committed:
INT-158: Comment updated.

Commit:
Adjust PHP formatting

Commit:
SOC-677 add unit tests

Commit:
SOC-677 update unit tests

Commit:
SOC-677 check if attributes in cache using attribute cache key, not preferences cache key

Commit:
CONCF-641 | remove Special:Mercury

harnash committed:
Added invalidateToken() to Helios/Client.

Grunny committed:
(MAIN-5155) Properly sanitize query when getting WF tags

Grunny committed:
(MAIN-5155) Properly sanitize query when getting WF tags

Commit:
added new right: editinterfacetrusted for more control over read/write JS rights

Commit:
Merge pull request #8103 from Wikia/CE-2480-allow-trusted-users-to-edit-js

added new right: editinterfacetrusted for read/write JS control

Commit:
Merge branch 'dev' of github.com:Wikia/app into dev

Commit:
Merge branch 'dev' into SERVICES-595

tomasznapieralski committed:
Merge pull request #8092 from Wikia/CONCF-978

CONCF-978 restrict permissions to getData and setData to admins only

Commit:
Add the Auth module

- Add the auth module to the DI framework
- Add logic for non existing users

Commit:
Add a test for not found

Commit:
Use authenticateViaTheSchwartz everywhere

- Use authenticateViaTheSchwartz everywhere
- Keep the responses consistent

tomasznapieralski committed:
Merge pull request #8088 from Wikia/CONCF-808

CONCF-808 | use one variable to specify large breakpoint

Commit:
Add a comment to HelperControllerWrapper

Commit:
Use AuthService instead of FQN

Commit:
newFromUser can return null

ArturKlajnerok committed:
Merge pull request #8098 from Wikia/SERVICES-595

Provide a hook for Helios to determine if a user is blocked

Commit:
Merge branch 'dev' into SOC-1153

Commit:
SOC-1070 Bail if there's no user title object and log an error

mari-mcmurtrie committed:
INT-158: Formatting change after CR. Also changed error message to be more appropreate.

mari-mcmurtrie committed:
Merge branch 'dev' into INT-158

Commit:
SOC-715 the new register & sign in buttons for the auth-nav

Commit:
SOC-715 wgEnableNewAuth is respected

Commit:
SOC-715 Account Nav opens login & reg pages in modal

Commit:
SOC-715 wgEnableNewAuth is respected in js also

bekcunning committed:
Merge pull request #8042 from Wikia/SOC-1153

SOC-1153 Use a flag to stop this task from sending a welcome message.

Commit:
Merge branches 'CONCF-967' and 'dev'

fraszczakszymon committed:
ADEN-2289 Renamed method name

fraszczakszymon committed:
ADEN-2289 Moved hidden class to template

bognix committed:
CONCF | fix permissions in CuratedContentController::getData

Commit:
Add a TODO

Commit:
Merge pull request #8109 from Wikia/CONCF-fix-permissions-for-get-data

CONCF | fix permissions in CuratedContentController::getData

fraszczakszymon committed:
ADEN-2289 Extracted modal handlers to new modules

Commit:
SOC-715 if a user is logged in, dropdown works as it shoud

Commit:
SOC-715 cursor pointer on the modal close button

dianafa committed:
Merge branch 'dev' into DAT-3082

Commit:
SOC-715 modal close button tracking

Commit:
ADEN-2335 use mappedVerticalName instead of wikiCategory

Commit:
ADEN-2335 use mappedVerticalName instead of wikiCategory

Commit:
ADEN-2335 removed fallback since we do not need it anymore

Commit:
ADEN-2335 removed fallback since we do not need it anymore

Grunny committed:
(CE-2485) Allow editing of JS pages for the editinterfacetrusted right

rafalkalinski commented on extensions/wikia/GlobalNavigation/templates/GlobalNavigationAccountNavigation_index.php:
```diff
@@ -1,24 +1,50 @@
+				<?php if ($isAnon) : ?>
+					<span class="auth-label register"><?= wfMessage( 'global-navigation-register' )->escaped(); ?></span>
+					<span class="auth-label"><?= wfMessage( 'global-navigation-or' )->escaped(); ?></span>
+					<span class="auth-label sign-in"><?= wfMessage( 'global-navigation-sign-in' )->escaped(); ?></span>
```
we need to prepare proper changes for that file when we will have some feedback from the designers about login & reg buttons behaviour

bkoval replied:
We need to have an i18n variable like "$1 or $2" where $1 and $2 are links to signup and login. Otherwise, this might get untranslateable.

bkoval replied:
Also, we need anchors instead of spans :)

fraszczakszymon committed:
ADEN-2337 Added Wikia vertical to mapping

fraszczakszymon committed:
ADEN-2337 Added Wikia vertical to mapping

Grunny committed:
Merge pull request #8113 from Wikia/CE-2485

(CE-2485) Allow editing of JS pages for the editinterfacetrusted right

fraszczakszymon committed:
ADEN-2337 Added test case

fraszczakszymon committed:
ADEN-2337 Added test case

Commit:
use invalidateToken to Helios/Client in UserLogout hook

hakubo committed:
Merge pull request #8099 from Wikia/CONCF-641

CONCF-641 | remove Special:Mercury

Grunny committed:
Merge branch 'dev' into phalanx-dont-show-blocker-for-article-comments

Conflicts:
	extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php

Grunny committed:
Minor whitespace tweaks per conventions

Grunny committed:
Merge pull request #7762 from WikiaVolDev/phalanx-dont-show-blocker-for-article-comments

VOLDEV-117

lukaszkonieczny committed:
CE-2429: escape shell args

dianafa committed:
fix the params array count issues

lukaszkonieczny committed:
CE-2429: escape just args

Commit:
ADEN-2339 $.exists() does not exist in regular jQuery

Commit:
Merge pull request #8117 from Wikia/aden-2339-fix-floor-adhesion-on-mercury

ADEN-2339 Fix floor adhesion ad unit on Mercury

Commit:
Merge pull request #8117 from Wikia/aden-2339-fix-floor-adhesion-on-mercury

ADEN-2339 Fix floor adhesion ad unit on Mercury

dianafa committed:
minor fixes

dianafa committed:
support multiple infoboxes

Commit:
SOC-1181 Ran stylize.php on all files in this extension

Commit:
SOC-1181 Remove uneeded code.

mari-mcmurtrie committed:
INT-158: improvement after CR

mari-mcmurtrie committed:
Merge branch 'dev' into INT-158

Commit:
SOC-677 remove calls to disableAutoload() in tests

Commit:
SOC-677 update setGlobalAttribute call to setGlobalFlag call

Commit:
Merge branch 'dev' into SOC-677

garthwebb committed:
Merge pull request #8120 from Wikia/SOC-1181

SOC-1181 Ran stylize.php on all files in this extension

dianafa committed:
return if no infoboxes

lukaszkonieczny committed:
CE-2429: Remove quotes

bkoval committed:
Abstract a method for building modal urls

bkoval committed:
Removing unneeded space

bkoval commented on extensions/wikia/GlobalNavigation/styles/GlobalNavigationAccountNavigation.scss:
```diff
@@ -73,14 +76,24 @@
 
-		&:hover {
+		&.register {
+			padding-left: 7px;
```
Why don't we move padding-left to a higher declaration so we avoid repeating it for .sign-in? :)

kvas-damian commented on extensions/wikia/AuthModal/css/AuthModal.scss:
```diff
@@ -61,6 +61,7 @@
 		background-position: center;
 		background-repeat: no-repeat;
 		box-sizing: content-box;
+		cursor: pointer;
```
Instead of hacking CSS maybe we could just add `href` to `a` tag?

bkoval replied:
:+1:

kvas-damian commented on extensions/wikia/GlobalNavigation/scripts/GlobalNavigationAccountNavigation.js:
```diff
@@ -10,6 +10,7 @@ require([
 	var $globalNavigation = $('#globalNavigation'),
 		loginAjaxForm = false,
 		$entryPoint;
+	;
```
?

kvas-damian commented on extensions/wikia/GlobalNavigation/scripts/GlobalNavigationAccountNavigation.js:
```diff
@@ -87,5 +87,30 @@ require([
+			});
+		}
+		else {
+			oldAccountNav();
```
What do you think about passing `$entryPoint` as a variable instead of using parent scope?

bkoval committed:
Adding unit tests for AuthComponent

kvas-damian commented on extensions/wikia/GlobalNavigation/styles/GlobalNavigationAccountNavigation.scss:
```diff
@@ -25,12 +25,15 @@
 		.links-container {
 			display: table-row;
 			height: $global-navigation-height-without-margin;
+			font-size: 12px;
```
The same font-size many times in the same file, what do you think about using variable?

kvas-damian commented on extensions/wikia/GlobalNavigation/styles/GlobalNavigationAccountNavigation.scss:
```diff
@@ -25,12 +25,15 @@
 			height: $global-navigation-height-without-margin;
+			font-size: 12px;
+			padding-top: 2px;
+			vertical-align: middle;
```
Isn't it just C&P from `.auth-label`?

bkoval commented:
Please, resolve merge conflicts by merging the newest SOC-904 :)

kvas-damian commented on extensions/wikia/GlobalNavigation/templates/GlobalNavigationAccountNavigation_index.php:
```diff
@@ -1,24 +1,50 @@
-		<div class="links-container<?php if ($isAnon): ?> anon <?php endif; ?>">
-			<?= $navItemLinkOpeningTag ?>
+<ul id="AccountNavigation" class="AccountNavigation table-cell <?= $enableNewAuth ? 'newAuth' : '' ?> ">
+	<?php if ($enableNewAuth): ?>
```
What do you think about separating new and old markup and put it in separate template files?

kvas-damian commented on extensions/wikia/GlobalNavigation/templates/GlobalNavigationAccountNavigation_index.php:
```diff
@@ -1,24 +1,50 @@
+				<? endif; ?>
+				<img class="chevron" src="data:image/gif;base64,R0lGODlhAQABAIABAAAAAP///yH5BAEAAAEALAAAAAABAAEAQAICTAEAOw%3D%3D"/>
+			</div>
+			<?php if ( !$isAnon ): ?>
```
Part for logged in user is duplicated, can you find a way to avoid code duplication?

rafalkalinski replied:
Yes. It waits for the designers guides about how should it work :)

dianafa committed:
Merge branch 'dev' into DAT-3082

kvas-damian commented:
Please resolve conflicts.

Commit:
Update PortableInfoboxClassification.php

CE-2367 change description

lukaszkonieczny committed:
Merge pull request #8073 from Wikia/CE-2367

CE-2367 infobox migration message in editor only for the insights list

Commit:
Merge pull request #8081 from Wikia/CE-2429

CE-2429: Migrate templates after adding new flag via Special:Flags

bognix committed:
Merge remote-tracking branch 'origin/dev' into CONCF-967

hakubo committed:
Use standard PHP function and log if videos is empty

hakubo committed:
remove logging, if no videos are found don't add them

vforge committed:
CONCF-967 | Fix CONCF-1065 and CONCF-1066

macbre committed:
PLATFORM-1317: set cURL options key by key

Commit:
Merge pull request #8125 from Wikia/PLATFORM-1317-set-options-using-curl_setopt

PLATFORM-1317: set cURL options key by key

vforge committed:
Merge pull request #8124 from Wikia/CONCF-908

CONCF-908 | Use standard PHP function, don't add videos when response is empty

dianafa committed:
fetch params from includeonly as well

Commit:
Remove facebookAppId from MercuryApi

Commit:
Merge remote-tracking branch 'origin/SOC-904' into SOC-715

dianafa committed:
refactor

vforge committed:
CONCF-967 | Fix item validation show

dianafa committed:
whitespace

Commit:
SOC-715 not using parent scope anymore

Commit:
SOC-715 fixed height of a modal

Commit:
Merge pull request #8029 from Wikia/CONCF-967

CONCF-967 | Remove check for duplicated title inside CC Desktop tool

Commit:
Merge branch 'dev' into SERVICES-464_Introduce_proper_MW_helios_client_calls

Commit:
Merge pull request #8089 from Wikia/aden-2289-hop-in-interstitial

ADEN-2289 Hop in interstitial

macbre committed:
Remove ExternalRubyCompiler for SASS

We use libsass PHP extension now, no need to spawn a Ruby process

bkoval committed:
Merge pull request #8100 from Wikia/SOC-904

SOC-904 AuthModal extension for New Auth Pages

Commit:
Merge pull request #8090 from Wikia/aden-2312-incontent-player-1x1

ADEN-2312 INCONTENT_PLAYER to 1x1 size

Commit:
Merge pull request #8114 from Wikia/aden-2337-corporate-pages-with-wka-wikia

ADEN-2337 Added Wikia vertical to mapping

Grunny committed:
Whitespace tweaks

Commit:
Merge pull request #8111 from Wikia/aden-2335-soisub3site-fix

ADEN-2335 SOI_SUB3SITE fix

Commit:
ADEN-2320 add wka.showcase

Commit:
ADEN-2320 add wka.showcase

Commit:
Merge remote-tracking branch 'origin/aden-2320-sudo-vertical' into aden-2320-sudo-vertical

Commit:
Back-ported fix for background from release-322

Merge pull request #8135 from Wikia/aden-2351-fix-wikia-skin
ADEN-2351 Fix for wikia skins on wide screens

Commit:
Merge pull request #8107 from Wikia/SOC-1070

SOC-1070 Bail if there's no user title object and log an error

mari-mcmurtrie committed:
Merge branch 'dev' into INT-158

dianafa committed:
hacky but works

mari-mcmurtrie committed:
Adding a note for clarification

Commit:
adding Wikia\Util\Optional and using in UserPreferences when getting defaults. also fixing logic in UserPreferences->prefIsSaveable, since that logic depended on a different part of the system deleting preferences

Commit:
simplify saving preferences to always set and not deal with the defaults and update tests.

Commit:
ramping swagger service usage to 5%

mari-mcmurtrie committed:
Merge pull request #8097 from Wikia/INT-158

INT-158: For Japanese Wikia on Mercury, no edit/add photo when anonymous edit is disabled.

Commit:
Revert "ramping swagger service usage to 5%"

This reverts commit 49c3d8d44cabb4e183072c67d4621e4ea7f9c17f.

Commit:
ADEN-2352 Initialize providers before first fillInSlot call

Commit:
Merge pull request #8130 from Wikia/PLATFORM-1313-libsass-cleanup

Remove ExternalRubyCompiler for SASS

macbre committed:
CurlHttpRequest: set a proxy to an empty string instead of null

Backport #8140

macbre committed:
CurlHttpRequest: cast proxy to a string to avoid passing a null value to cURL

Commit:
Merge pull request #8141 from Wikia/PLATFORM-1317-cast-proxy-to-a-string

CurlHttpRequest: cast proxy to a string to avoid passing a null value to cURL

dianafa committed:
handle includeonly tag

macbre committed:
PLATFORM-1317: only set when the proxy is not an empty value

Follow the convention used in PhpHttpRequest class

Backporting #8142

lukaszkonieczny committed:
CE-2499: Fix for Insights Query Pages

vforge committed:
CONCF-1008 | Do not save empty sections (without items)

Commit:
ADEN-2320 moving querysting instance creation upper

Grunny committed:
Merge pull request #8143 from Wikia/CE-2499

CE-2499: Fix for Insights Query Pages

bkoval committed:
Simplifying template

bkoval committed:
--amend

bkoval committed:
Adding hover state to links

Commit:
Merge pull request #8147 from Wikia/SOC-601

SOC-601 | Banner notification on successful account registration

Commit:
Add User::isCurrent() method

This method is loosely inspired by User::equals() from MW 1.25
and is meant to replace all the Id / Name comparisons the context
of which would be to determine whether the user in question is
the current performer.

Commit:
Merge pull request #8148 from Wikia/dev-add-user-iscurrent-method

Add User::isCurrent() method

bkoval committed:
Simplifying SCSS for auth links in global nav

bkoval committed:
Further scss simplifications

bkoval committed:
Adding href=# instead of cursor:pointer

Commit:
use DI to inject helios client

hakubo committed:
remove logging, if no videos are found don't add them

This method got removed in CONCF-908
as it could be changed to standard PHP function
I have missed to change it here as well

Commit:
Merge pull request #8126 from Wikia/MercuryApi-update

Remove facebookAppId from MercuryApi

hakubo committed:
Merge pull request #8150 from Wikia/MAIN-5181

Use standard PHP function instead of custom one

dianafa committed:
extract to getParsedInfoboxes function

bognix committed:
CONCF-1046 fix empty image url situation

bognix committed:
CONCF-1046 fix PHP errors

dianafa committed:
replace str_replace with preg_match_all

Commit:
Merge branch 'dev' into SERVICES-464_Introduce_proper_MW_helios_client_calls

bognix committed:
CONCF-1046 add unit tests

Commit:
fix Dependency Injection documentation and usage in Helios UserTest

Commit:
remove isPresent and use null

Commit:
adding tests for Optional

Commit:
SOC-577 consolidate logging into method, remove thrown exception from AttributeKeyValueService::delete

nmonterroso committed:
Merge pull request #8137 from Wikia/SERVICES-639

Don't Filter Preferences When Saving

Commit:
SOC-677 use === when comparing attributes, set/delete values in service before cache

Commit:
SOC-677 update @return in method docs

Commit:
SOC-677 fix unit tests

dianafa committed:
move logic to DataService

Commit:
SOC-1186 Make sure we don't set wgDBname

Doing this has weird effects on DB handle loading from wgGetDB()

Also did some clean up of WikiService::getWikiAdminIds

Commit:
SOC-1186 Spacing fix

Commit:
SOC-677 remove commented out test

Commit:
SOC-1186 Make sure we keep the same behavior as previous

Commit:
setting user preferences service to 5%

Commit:
SOC-1186 Make this explicitly a static function

Commit:
Merge branch 'dev' into SOC-1007

garthwebb committed:
Merge pull request #7828 from Wikia/SOC-1007

SOC-1007 Fix a null assignment error

Commit:
SOC-677 use is_null() instead of empty() when checking if attribute not set in cache

Commit:
Merge pull request #8121 from Wikia/SOC-677

SOC-677 Start writing attributes to Attribute Service

dianafa committed:
add namespace check

dianafa committed:
fix check

dianafa committed:
add NS_TEMPLATE

vforge committed:
CONCF-1008 | Add some explanation

bognix committed:
CONCF-1046 rename variable

Commit:
SOC-715 proper hover color for the global nav buttons

Commit:
SOC-715 global nav login & reg buttons enabled

Commit:
SOC-1188 | Disable cache for FB connect

Commit:
Merge pull request #8139 from Wikia/aden-2352-providers-init

ADEN-2352 Initialize providers before first fillInSlot call

Commit:
SOC-1188 | Disable cache for FB connect - fix unit tests

Commit:
SOC-715 upper z-index for the modal

Commit:
Merge pull request #8134 from Wikia/aden-2320-sudo-vertical

ADEN-2320 add wka.showcase

bkoval committed:
Using one message to parse auth options

bkoval committed:
Ensure the right distance between avatar and links

idradm committed:
DAT-3114 add mw api for acquiring all infoboxes list

bognix committed:
Merge remote-tracking branch 'origin/dev' into CONCF-1046

tomasznapieralski committed:
Merge pull request #8152 from Wikia/CONCF-1046

Concf 1046 fix empty image url when image id points to non-file page

macbre committed:
UserAvatarsApi: add Swagger-generated classes

macbre committed:
Stylize the code to match MW conventions

macbre committed:
Remove debugging code

Commit:
ADEN-2320 CR remarks

bkoval committed:
Adding qqq

Commit:
ADEN-2320 CR remarks

macbre committed:
Remove User/Avatars/Model

bkoval committed:
Adding crowdin config file

macbre committed:
ApiClient: no need to generate a diff here

vforge committed:
CONCF-1008 | Add basic unitest

vforge committed:
Merge remote-tracking branch 'origin/dev' into CONCF-1008

Commit:
Merge pull request #8161 from Wikia/PLATFORM-1334-add-UserAvatarsApi-class

UserAvatarsApi: add Swagger-generated classes

Commit:
Merge remote-tracking branch 'origin/dev' into SOC-715

idradm committed:
DAT-3114 add query page for all infoboxes

Commit:
Merge pull request #8162 from Wikia/aden-2320-sudo-vertical

ADEN-2320 CR remarks

Commit:
Merge pull request #8115 from Wikia/SERVICES-464_Introduce_proper_MW_helios_client_calls

Services-464 add helios/client invalidate token call in UserLogout hook

bkoval committed:
Changing name of the crowdin project id to the one created in crowdin

dianafa committed:
fix includeonly issues

dianafa committed:
add isset and if

idradm committed:
DAT-3114 add consts, make var names more meaningful

vforge committed:
CONCF-1008 | Separate function, add unittest

Commit:
Merge pull request #8160 from Wikia/SOC-1188

SOC-1188 | Disable cache for FB connect

dianafa committed:
add

nikodamn committed:
Merge pull request #8128 from Wikia/DAT-3082

DAT-3082 show all params used in infoboxes

nikodamn committed:
Merge pull request #8163 from Wikia/DAT-3114

Dat 3114 create all infoboxes api and query page

Commit:
SOC-1186 Better way of representing IN clauses

garthwebb committed:
Merge pull request #8156 from Wikia/SOC-1186

SOC-1186 Make sure we don't set wgDBname

Commit:
replacing getOption calls

Commit:
bumping user pref service to 15%

Commit:
SOC-715 fixes for the global nav login dropdown

nmonterroso committed:
Merge pull request #8166 from Wikia/pref-service-ramp

Pref service ramp

Commit:
update User::setGlobalAttribute calls to User::setGlobalFlag calls

Commit:
Merge pull request #8167 from Wikia/update-setglobalflag-call

update User::setGlobalAttribute calls to User::setGlobalFlag calls

Commit:
SOC-715 links for the reg & login global nav buttons

tomasznapieralski committed:
Merge pull request #8146 from Wikia/CONCF-1008

CONCF-1008 | Do not save empty sections (without items)

Commit:
SOC-715 anon avatar link fixed

Commit:
bumping user pref service to 25%

ArturKlajnerok committed:
Merge pull request #8169 from Wikia/pref-service-ramp

bumping user pref service to 25%

Commit:
SOC-715 anon avatar link improved

Commit:
Merge remote-tracking branch 'origin/dev' into SOC-715
